include/stl2/detail/iterator/any_iterator.hpp

```diff
@@ -0,0 +1,230 @@
+                ::new (static_cast<void *>(&data_.tiny)) I(std::move(i));
+                deref_ = &deref_small<Reference, I>;
+                exec_ = &exec_small<RValueReference, I>;
+            }
+            InputIterator{I} cursor(I i, big_tag) {
+                data_.big = new shared_iterator<I>(std::move(i));
+                deref_ = &deref_big<Reference, I>;
+                exec_ = &exec_big<RValueReference, I>;
+            }
+        public:
+            using value_type = ValueType;
+            using single_pass = std::true_type;
+            struct mixin;
+            cursor() = default;
+            cursor(cursor &&that) {
+                that.exec_(op::move, &that.data_, &data_);
+                swap(deref_, that.deref_);
+                swap(exec_, that.exec_);
+            }
+            cursor(cursor const &that) {
+                that.exec_(op::copy, const_cast<blob *>(&that.data_), &data_);
+                deref_ = that.deref_;
+                exec_ = that.exec_;
+            }
+            cursor &operator=(cursor &&that) {
+                if (&that != this) {
+                    this->~cursor();
+                    ::new (static_cast<void *>(this))
+                        cursor(std::move(that));
+                }
+                return *this;
+            }
+            cursor &operator=(cursor const &that) {
+                if (&that != this) {
+                    this->~cursor();
+                    ::new (static_cast<void *>(this)) cursor(that);
+                }
+                return *this;
+            }
+            ~cursor() {
+                exec_(op::nuke, &data_, nullptr);
+            }
+            Reference read() const {
+                return deref_(data_);
+            }
+            bool equal(cursor const &that) const {
+                return exec_(op::comp, const_cast<blob *>(&data_),
+                    const_cast<blob *>(&that.data_)) != nullptr;
+            }
+            void next() {
+                exec_(op::bump, &data_, nullptr);
+            }
+            RValueReference indirect_move() const {
+                return exec_(op::rval, nullptr, nullptr)(data_);
+            }
+        };
+
+        template<class Reference, class ValueType, class RValueReference>
+        struct cursor<Reference, ValueType, RValueReference>::mixin
+        : basic_mixin<cursor> {
+            mixin() = default;
+            using mixin::basic_mixin::basic_mixin;
+            InputIterator{I} mixin(I i)
+            requires
+                ConvertibleTo<reference_t<I>, Reference>() &&
+                ConvertibleTo<rvalue_reference_t<I>, RValueReference>()
+            : basic_mixin<cursor>(cursor{std::move(i), is_small<I>{}}) {
+            }
+            template <class I, class C = cursor,
+                class = meta::if_c<!is_same<I, mixin>::value>>
+            requires
+                !Same<I, basic_iterator<C>>() && InputIterator<I>() &&
```

Comment:
It would be interesting to know why I needed to use SFINAE here (`meta::if_c<!is_same...>`) instead of concepts (`!Same<...`). Also interesting about this: I couldn't form `basic_iterator<cursor>` here because `basic_iterator`'s template parameter is required to satisfy `Cursor`, and concept checking gets into some weird recursive instantiation SNAFU.